### DISCUSSION TITLE
You should support tiered pricing and usage based pricing

### DISCUSSION THREAD

**Discussion by dfrankle2** [Glench/ExtPay#103] [Ideas]
This would be very helpful for me!

**Comment by Glench** [comment]
Thanks for the suggestion! Can you please explain in detail what your use case is and what you envision?

**Reply by rurimiao** [reply]
I have the same problem. It would be set some additional purchases for additional features, especially some discrete functions required by different users group, like some advanced panels, actions, skins, themes, etc.

**Reply by clark-teeple** [reply]
@Glench I think they meant like enable/disable features based on subscription tier, or limit API calls to some set number based on subscription tier or a credit count, or something like that. I would also be interested in this.

**Comment by JPaulHendricks** [comment]
The simple scenario is 1) Free Tier, 2) Premium Tier (lower price) and 3) Pro Tier (higher price).

**Comment by Glench** [comment]
This is coming! We won't give timing, but it's in active development and is our highest priority.

**Reply by ba32107** [reply]
Hi @Glench, we understand you don't want to commit to a date, but it would be very helpful for me (and I imagine a lot of people) if you could give us a very rough estimate. Like, can we expect this in a few months? By Q1 next year? More than a year from now? It doesn't have to be more precise than that.

I now have to decide how to implement the payment integration in my extension and knowing when the multi-tier feature arrives is a critical piece of info to how I schedule my next few months of work.

**Reply by Glench** [reply]
OK sure. Hopefully it will be done by the end of year, but possibly much sooner.

**Reply by ba32107** [reply]
Thanks!

**Reply by pizzaisdavid** [reply]
@Glench Any update about this in any regards? "this" referring to having "tiers" like free, premium, super-premium.

It would help me with some planning for my software.

Thanks

**Reply by Glench** [reply]
Hoping to start testing "this" by the end of the year. If you want to be a beta tester please let me know.

**Reply by pizzaisdavid** [reply]
I'm definitely open to the idea, would need more details. I'd probably want to test on my non-production build within my app, which could probably be achieved with just doing a different project id when in development mode.

**Reply by marc-pls** [reply]
@Glench I would also love to be apart of your beta testers if this is still an option. I'm looking to launch my app with monthly, yearly and lifetime pricing. Would "this" fall into the tiered structure being implemented?

**Reply by ba32107** [reply]
Hi @Glench, do you have an update on this? I am just about to start integrating the payment system into my extension - do you have an estimate on when this feature will be production ready?

**Reply by Glench** [reply]
Still working on it, slowly but surely! Sorry for the missed estimate. We're about 95% done with the beta and then will test with a few folks to make sure everything works. We try not to give estimates (like I did above) because there are lots of unknowns in the development process and with our personal schedules, but it will be done eventually!

**Reply by Vishaltyagi700** [reply]
@Glench Thanks for the awesome work, do we have any visibility on this. I am considering Extension Pay and am eying multiple plans suport.

**Reply by james-arn** [reply]
I'm looking for tiered and usage pricing too, is the deciding factor for me between using extpay or not, it looks like it's been in the works for 1.5 years now with this not rolled out?

**Reply by Vishaltyagi700** [reply]
Hey we love the work, please let us know any ETAs. You've been MIA, hope things have been well.

**Reply by Vishaltyagi700** [reply]
let us know if we can be of any help and contribution to the product in any which way possible.

**Reply by james-arn** [reply]
To the person who replied to me saying 'f*** your entitlement, it will be ready when it's ready" with my comment:

I wasnt trying to be rude or entitled, I was trying to explain I'm on the cusp of either setting up this (paid) product or not, I can't use it if this feature isn't there, so it's feedback for demand for the tool and probably more money for them if implemented. They may also of replied saying they would be launching it this week so I would if held out.

**Reply by bilalsattar24** [reply]
I'm absolutely shocked that the owner of this business said that to a potential customer. I was confused and thought it was a hack or a troll account.

**Reply by Glench** [reply]
Yeah, sorry guys. It was a stupid thing to say. I've been struggling with several very challenging health and mental health challenges and the stress that comment triggered in me came out as defensiveness. Thanks for any understanding you can generate.

The good news is that I actually *have* been working on the next version of ExtPay and it's almost done! I'm just finishing up the documentation now. This version includes multiple plans, coupon codes, more payment methods for higher revenue, and a better dev experience. It's also backward compatible so you should just be able to drop it into existing extensions.

It's been a huge endeavor that's touched every part of the existing system which is why it's taken so long. I and several beta testers have been using it in production for 6 months so I'm hoping it's ready to go but I'm a little nervous about releasing it. I'll let this thread know when it's out but I expect it will be pretty soon!

**Reply by complanboy2** [reply]
Hey Glen, 

Thank you for all the great work. 
I'm a beta tester and it's working wonderful so far. Looking forward to integrate into existing one's. Any date we can look forward?

**Reply by Glench** [reply]
Haha thanks for the review. 

> Any date we can look forward?

Yes! My plan is to release it this Saturday morning US/Eastern time!

**Reply by Vishaltyagi700** [reply]
I am excited to try it, Thanks Glench

**Reply by Glench** [reply]
It's out! 3.1.0 is live. Check out the changelog and latest README!

**Comment by Glench** [comment]
ExtPay version 3.1.0 is out now and includes multiple plans and lots of other stuff! Check it out!
